sandy/core/cov.py

```diff
@@ -1444,17 +1442,34 @@ def random_corr(cls, size, correlations=True, seed=None, **kwargs):
         return cls(corr, **kwargs)
 
     @classmethod
-    def random_cov(cls, size, stdmin=0.0, stdmax=1.0, correlations=True,
-                   seed=None, **kwargs):
+    def random_cov(cls, size, seed=None):
         """
+        Construct a covariance matrix with random values
+
+        Parameters
+        ----------
+        size : `int`
+            Dimension of the original matrix
+        seed : `int`, optional, default is `None`
+            seed for the random number generator (by default use `numpy`
+            dafault pseudo-random number generator)
+
+        Returns
+        -------
+        `CategoryCov`
+            object containing covariance matrix
+
+        Examples
+        --------
         >>> sandy.CategoryCov.random_cov(2, seed=1)
                     0           1
-        0 2.15373e-02 5.97134e-03
-        1 5.97134e-03 8.52642e-03
+        0 4.59962e-02 4.58318e-02
+        1 4.58318e-02 4.56679e-02
         """
-        corr = random_corr(size, correlations=correlations, seed=seed)
-        std = np.random.uniform(stdmin, stdmax, size)
-        return cls.corr2cov(corr, std, **kwargs)
+        np.random.seed(seed)
+        matrix = scipy.random.rand(size, size)

```

Comment:
I understand the concept, but then how do you decide the size of the sample.
Have a good look at np.cov to make sure you really know what you are doing